### PR TITLE
Deprecated initialCaps

### PR DESCRIPTION
**Pull request by wing328**
### PR checklist

- [ ] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [ ] Ran the shell script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh` and `./bin/security/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates). Windows batch files can be found in `.\bin\windows\`.
- [ ] Filed the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`~~, `3.4.x`, `4.0.x`~~. Default: `master`.
- [ ] Copied the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

### Description of the PR

Replacing it with `camelize` and will completely remove it in 4.x release

cc @OpenAPITools/generator-core-team

### CODE REVIEW

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultCodegen.java [hunk 1]
```diff
@@ -1505,14 +1505,15 @@ public String snakeCase(String name) {
...
+     * Capitalize the string. Please use org.openapitools.codegen.utils.StringUtils.camelize instead as this method will be deprecated.
      *
      * @param name string to be capitalized
      * @return capitalized string
+     * @deprecated
      */
     @SuppressWarnings("static-method")
     public String initialCaps(String name) {
-        return StringUtils.capitalize(name);
+        return camelize(name);
```
Comment (osharav) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultCodegen.java]: camelize removes underscores which may cause issues with the InlineModelResolver and other places where underscores are being introduced. in the cpp rest sdk client generator this causes the class names to be different than the file names (some method calls have used initialCaps and some do not) which causes import errors.
I will open a dedicated issue for this.
Regards,
Omer.

**Comment by osharav** [comment]
#2003 cloud be related

**Comment by wing328** [comment]
@osharav thanks for reviewing the change. I could be wrong but I don't think #2002 is related as the user has confirmed 3.4.4 (which does not include this change) also has the issue.

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultCodegen.java (hunk 1 above)
Comment (wing328) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultCodegen.java]: Yes, please open a separate issue with a (minimal) spec to reproduce the issue.

**Comment by osharav** [comment]
True, but I have another example with a generated inline response which shows the behavior - I'll attach it to the issue I need to open.

**Pull request merged**